src/browser/filestorage.js

```diff
@@ -99,10 +99,14 @@ ServerFileStorageWrapper.prototype.load_from_server = function(sha256sum)
 {
     return new Promise((resolve, reject) =>
     {
-        v86util.load_file(this.baseurl + sha256sum, { done: buffer =>
+        v86util.load_file(this.baseurl + sha256sum, { done: async buffer =>
         {
             const data = new Uint8Array(buffer);
-            this.cache(sha256sum, data).then(() => resolve(data));
+            await this.cache(sha256sum, data);
+
+            return new Promise((resolve,reject)=>{
+                resolve(data);
+            });
```

Comment:
I think this can be simplified to `return Promise.resolve(data);` （?

Reply:
@undefined-moe i have modified this piece of code.